File: mobile/src/main/java/com/windscribe/mobile/generalsettings/GeneralSettingsView.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.generalsettings;

public interface GeneralSettingsView {

    String[] getOrderList();

    String[] getThemeList();

    void openFileChooser(int requestCode);

    void registerLocaleChangeListener();

    void resetTextResources(String title, String sortBy, String latencyDisplay, String language, String appearance,
            String notificationState, String hapticFeedback, String version, String connected, String disconnected,
            String appBackground);

    void setActivityTitle(String activityTitle);

    void setAppVersionText(String versionText);

    void setConnectedFlagPath(String path);

    void setDisconnectedFlagPath(String path);

    void setFlagSizeLabel(String label);

    void setLanguageTextView(String language);

    void setLatencyType(String latencyType);

    void setSelectionTextView(String selection);

    void setThemeTextView(String theme);

    void setupCustomFlagAdapter(String saved, String[] options);

    void setupHapticToggleImage(int ic_toggle_button_off);

    void setupLanguageAdapter(String savedLanguage, String[] language);

    void setupLatencyAdapter(String savedLatency, String[] latencyTypes);

    void setupLocationHealthToggleImage(int image);

    void setupNotificationToggleImage(int ic_toggle_button_off);

    void setupSelectionAdapter(String savedSelection, String[] selections);

    void setupThemeAdapter(String savedTheme, String[] themeList);
}

File: mobile/src/main/java/com/windscribe/mobile/generalsettings/GeneralSettingsView.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.generalsettings

interface GeneralSettingsView {
    val orderList: Array<String>
    val themeList: Array<String>
    fun openFileChooser(requestCode: Int)
    fun registerLocaleChangeListener()
    fun resetTextResources(
        title: String,
        sortBy: String,
        latencyDisplay: String,
        language: String,
        appearance: String,
        notificationState: String,
        hapticFeedback: String,
        version: String,
        connected: String,
        disconnected: String,
        appBackground: String
    )

    fun setActivityTitle(activityTitle: String)
    fun setAppVersionText(versionText: String)
    fun setConnectedFlagPath(path: String)
    fun setDisconnectedFlagPath(path: String)
    fun setFlagSizeLabel(label: String)
    fun setLanguageTextView(language: String)
    fun setLatencyType(latencyType: String)
    fun setSelectionTextView(selection: String)
    fun reloadApp()
    fun setupCustomFlagAdapter(
        localiseValues: Array<String>,
        selectedKey: String,
        keys: Array<String>
    )

    fun setupHapticToggleImage(ic_toggle_button_off: Int)
    fun setupLanguageAdapter(
        localiseValues: Array<String>,
        selectedKey: String,
        keys: Array<String>
    )

    fun setupLatencyAdapter(
        localiseValues: Array<String>,
        selelctedKey: String,
        keys: Array<String>
    )

    fun setupLocationHealthToggleImage(image: Int)
    fun setupNotificationToggleImage(ic_toggle_button_off: Int)
    fun setupSelectionAdapter(
        localiseValues: Array<String>,
        selectedKey: String,
        keys: Array<String>
    )

    fun setupThemeAdapter(localiseValues: Array<String>, selectedKey: String, keys: Array<String>)
}